### DISCUSSION TITLE
Problem with e2e tests

### DISCUSSION THREAD

**Discussion by pahan35** [storybookjs/storybook#16697] [General] [answered]
I noticed that you have broken e2e tests.

Actually, there are 2 broken tests.
1. `should have a dark background` in `addon-backgrounds` in file `addon-backgrounds.spec.ts`
![image](https://user-images.githubusercontent.com/13823215/141974525-9a471835-78a5-4662-ad06-c658100159d4.png)
2. `should provide source snippet` in `addon-docs` in file `addon-docs.spec.ts`.
![image](https://user-images.githubusercontent.com/13823215/141974322-1ae71d24-0abe-4169-a5b9-f92927f278c2.png)

The 2nd one looks like an easy fix: just need to adjust the match pattern, if the expected output is changed. Can somebody clarify what should we do here?

The 1st one is a bit unclear. It should have the expected value, but for some reason test fails. Does anybody have any idea how to get access to the recorded video and the screenshot? They are mentioned in the report log.

**Comment by shilman** [comment, marked as answer]
Thanks for raising this @pahan35. I believe @gaetanmaisse has been looking into this.

1) I have no idea what's going on in the first issue -- needs debugging. Maybe something to do with the major rearchitecture of core in 6.4?
2) The second issue is a bug and I'm probably best suited to track it down. There are two types of source snippets in addon-docs: static snippets that come from the original source code of the story, and dynamically-generated snippets that are generated from the output result of the story function. The test says that it should NOT be the former, but somehow it is in this case.

As for the artifacts, you should be able to access them in the Circle CI run:

<img width="1272" alt="e2e-tests-pnp__352432__-_storybookjs_storybook" src="https://user-images.githubusercontent.com/488689/141978336-c3eba25f-af0d-4763-95b8-fa7bb2627348.png">

**Reply by pahan35** [reply]
I've reproduced 1st bug in my environment. I'll find a way how to fix it.

P.S: with artifacts, it's much easier. Thanks for your guidance!

**Reply by pahan35** [reply]
It looks like the `theme` is not updated when `PreviewContainer` is rendered. https://github.com/storybookjs/storybook/blob/ded097e07365cbea464e832c244563914d6c4d59/lib/components/src/blocks/Preview.tsx#L100

https://github.com/storybookjs/storybook/blob/ded097e07365cbea464e832c244563914d6c4d59/lib/components/src/blocks/BlockBackgroundStyles.tsx#L3-L9

**Reply by pahan35** [reply]
@gaetanmaisse please let me know if I can help with any of those bugs.
